src/outputProcessor/HtmlOutputProcessor.ts

```diff
@@ -0,0 +1,181 @@
+  let {typeColor, icon, messageColor, trim = options.defaultTrimLength} = messageConfig;
+
+  if (severity === 'error') {
+    typeColor = COLORS.RED;
+    icon = LOG_SYMBOLS.ERROR;
+  } else if (severity === 'warning') {
+    typeColor = COLORS.YELLOW;
+    icon = LOG_SYMBOLS.WARNING;
+  }
+
+  const maybeTrimLength = (msg: string) =>
+    trim && msg.length > trim
+      ? msg.substring(0, trim) + ' ...\n\n... remainder of log omitted ...\n'
+      : msg;
+
+  const processMessage = (msg: string) => escapeHtml(maybeTrimLength(msg));
+
+  if (type == 'cy:log') {
+    processedMessage = utils.applyMessageMarkdown(processedMessage, {
+      bold: (str) => `<b>${str}</b>`,
+      italic: (str) => `<i>${str}</i>`,
+      processContents: processMessage,
+    });
+  } else {
+    processedMessage = processMessage(processedMessage);
+  }
+
+  // If the message is multilined, align non-first lines with the "message column" that's
+  // to the right of the "type column"
+  processedMessage = processedMessage
+    .split('\n')
+    .map((line, index) => (index === 0 ? line : `${' '.repeat(TYPE_STR_PADDING + 1)}${line}`))
+    .join('\n');
+
+  processedMessage = `<pre${messageColor ? ` style="color:${messageColor};"` : ''}>${processedMessage}</pre>`;
+
+  const typeString = `<pre style="color:${typeColor};">${
+    // pad to make "type column" spacing even
+    `${type}${icon}`.padStart(TYPE_STR_PADDING, ' ')
+  }</pre>`;
+
+  return `<p>${typeString} ${processedMessage}</p>\n`;
+}
+
+/**
+ *  Custom html output processor intended to be used with
+ * `cypress-terminal-report`. Generates detailed html log files that log
+ *  every Cypress message in a test-- in other words, log exactly what is
+ *  in the messages window when running `cypress open`)
+ *
+ *  Example usage in config file:
+ *  ```
+ * import installTerminalReporter from
+ *    'cypress-terminal-report/src/installLogsPrinter'
+ *
+ * installTerminalReporter(on, {
+      printLogsToFile: 'always',
+      outputRoot: config.reporterOptions.resultsDir,
+      specRoot: config.reporterOptions.testsDir,
+      outputTarget:()=>({ 'logs|html': detailedLogHtmlOutputProcessor }),
+    })
+    ```
+ */
+export default class HtmlOutputProcessor extends BaseOutputProcessor {
+  write(allMessages: AllMessages) {
+    for (const [spec, tests] of Object.entries(allMessages)) {
+      let html = `<h1>${escapeHtml(spec)}</h1>\n`;
+
+      for (const [test, messages] of Object.entries(tests)) {
+        html += `<h2>${escapeHtml(test)}</h2>\n`;
+        for (const message of messages) {
+          html += formatMessage(message, this.options);
+        }
+        html += '<br>\n';
+      }
+
+      html = `<html>
+<head>
+  <style>
+    body { font-family: monospace; }
+    p { margin: 0; padding: 0; }
+    pre { display: inline; margin: 0; }
+    h2 { margin: 0; font-size: 1.2em; }
+  </style>
+</head>
+<body>
+  ${html}
+</body>
+</html>`;
+
+      this.writeSpecChunk(spec, html);
```

Comment:
I think this might not be correct. Multiple chunks will be written, and you will have to handle as prefix what is before ${html} and as suffix what is after ${html}. Currently I believe it would have duplicate <html> tags.

Reply:
Yep you were right! Got that fixed.